nb/projects/belt_sprockets/belt_sprockets.nb

```diff
@@ -0,0 +1,683 @@
+    [0.863705, 0.408181, 0],
+    [0.90364, 0.217664, 0],
+    [0.932921, 0.130969, 0],
+    [0.988924, 0.062001, 0],
+    [1.065168, 0.016448, 0],
+    [1.155171, 0, 0],
+    [1.155171, -0.5, 0],
+  ];
+
+  const GT2_5mm = [
+    [-1.975908, -0.75, 0],
+    [-1.975908, 0, 0],
+    [-1.797959, 0.03212, 0],
+    [-1.646634, 0.121224, 0],
+    [-1.534534, 0.256431, 0],
+    [-1.474258, 0.426861, 0],
+    [-1.446911, 0.570808, 0],
+    [-1.411774, 0.712722, 0],
+    [-1.368964, 0.852287, 0],
+    [-1.318597, 0.989189, 0],
+    [-1.260788, 1.123115, 0],
+    [-1.195654, 1.25375, 0],
+    [-1.12331, 1.380781, 0],
+    [-1.043869, 1.503892, 0],
+    [-0.935264, 1.612278, 0],
+    [-0.817959, 1.706414, 0],
+    [-0.693181, 1.786237, 0],
+    [-0.562151, 1.851687, 0],
+    [-0.426095, 1.9027, 0],
+    [-0.286235, 1.939214, 0],
+    [-0.143795, 1.961168, 0],
+    [0, 1.9685, 0],
+    [0.143796, 1.961168, 0],
+    [0.286235, 1.939214, 0],
+    [0.426095, 1.9027, 0],
+    [0.562151, 1.851687, 0],
+    [0.693181, 1.786237, 0],
+    [0.817959, 1.706414, 0],
+    [0.935263, 1.612278, 0],
+    [1.043869, 1.503892, 0],
+    [1.123207, 1.380781, 0],
+    [1.195509, 1.25375, 0],
+    [1.26065, 1.123115, 0],
+    [1.318507, 0.989189, 0],
+    [1.368956, 0.852287, 0],
+    [1.411872, 0.712722, 0],
+    [1.447132, 0.570808, 0],
+    [1.474611, 0.426861, 0],
+    [1.534583, 0.256431, 0],
+    [1.646678, 0.121223, 0],
+    [1.798064, 0.03212, 0],
+    [1.975908, 0, 0],
+    [1.975908, -0.75, 0],
+  ];
+
+  const pulley = (belt_type, pulley_OD, tooth_depth, tooth_width) => {
+    const tooth_distance_from_centre = Math.sqrt(
+      Math.pow(pulley_OD / 2, 2) -
+        Math.pow((tooth_width + additional_tooth_width) / 2, 2)
+    );
+    const tooth_width_scale =
+      (tooth_width + additional_tooth_width) / tooth_width;
+    const tooth_depth_scale =
+      (tooth_depth + additional_tooth_depth) / tooth_depth;
+
+    const center = Arc(pulley_OD, pulley_OD);
+
+    //teeth - cut out of shaft
+
+    var toothProfile = [];
+    if (profile == 1) {
+      toothProfile = MXL;
+    }
+    if (profile == 2) {
+      toothProfile = DP40;
+    }
+    if (profile == 3) {
+      toothProfile = XL;
+    }
+    if (profile == 4) {
+      toothProfile = H;
+    }
+    if (profile == 5) {
+      toothProfile = T2_5;
+    }
+    if (profile == 6) {
+      toothProfile = T5;
+    }
+    if (profile == 7) {
+      toothProfile = T10;
+    }
+    if (profile == 8) {
+      toothProfile = AT5;
+    }
+    if (profile == 9) {
+      toothProfile = HTD_3mm;
+    }
+    if (profile == 10) {
+      toothProfile = HTD_5mm;
+    }
+    if (profile == 11) {
+      toothProfile = HTD_8mm;
+    }
+    if (profile == 12) {
+      toothProfile = GT2_2mm;
+    }
+    if (profile == 13) {
+      toothProfile = GT2_3mm;
+    }
+    if (profile == 14) {
+      toothProfile = GT2_5mm;
+    }
+
+    //Convert path to a face
+    toothProfile = Shape.fromOpenPath(toothProfile).close().fill();
```

Comment:
I'd consider moving this to the profile definitions, then you can just pass toothProfile in directly, and we can have a view of each profile in the notebook